base/compiler/tfuncs.jl

```diff
@@ -2213,9 +2218,9 @@ function getfield_effects(arginfo::ArgInfo, @nospecialize(rt))
     if !(length(argtypes) ≥ 3 && getfield_notundefined(obj, argtypes[3]))
         consistent = ALWAYS_FALSE
     end
-    nothrow = getfield_nothrow(arginfo, :on)
+    bcheck = getfield_boundscheck(arginfo)
+    nothrow = getfield_nothrow(𝕃, arginfo, bcheck)
```

Comment:
I'm not sure why we gave hard-coded `:on` for this `getfield_nothrow` here. @Keno was it intentional?